Framework/Algorithms/test/IntegrationTest.h

```diff
@@ -146,15 +146,31 @@ class IntegrationTest : public CxxTest::TestSuite {
     Workspace2D_sptr input;
     TS_ASSERT_THROWS_NOTHING(
         input = std::dynamic_pointer_cast<Workspace2D>(AnalysisDataService::Instance().retrieve("testSpace")))
-    assertRangeWithPartialBins(input);
+    assertRangeWithPartialBins(input, 0.1, 4.5, {52., 74., 96.}, {6.899, 8.240, 9.391});
   }
 
   void testRangeWithPartialBinsAndDistributionData() {
     Workspace2D_sptr input;
     TS_ASSERT_THROWS_NOTHING(
         input = std::dynamic_pointer_cast<Workspace2D>(AnalysisDataService::Instance().retrieve("testSpace")))
     input->setDistribution(true);
-    assertRangeWithPartialBins(input);
+    assertRangeWithPartialBins(input, 0.1, 4.5, {52., 74., 96.}, {6.899, 8.240, 9.391});
+  }
+
+  void testRangeWithPartialBinsAndLimitsInSameBin() {
+    Workspace2D_sptr input;
+    TS_ASSERT_THROWS_NOTHING(
+        input = std::dynamic_pointer_cast<Workspace2D>(AnalysisDataService::Instance().retrieve("testSpace")))
+    input->setDistribution(true);
+    assertRangeWithPartialBins(input, 0.1, 0.5, {4., 6., 8.}, {1.265, 1.549, 1.788});
+  }
+
+  void testRangeWithPartialBinsAndEqualLimits() {
```

Comment:
New tests look good, but is it also worthwhile having a test that asserts integration on a points and histogram data set that are equivalent give the same answer? It doesn't look like we have a test that does this but I may have missed it.